src/plugins/embeddable/public/lib/panel/panel_header/panel_header.tsx

```diff
@@ -68,7 +68,14 @@ function renderNotifications(
     const context = { embeddable };
 
     let badge = notification.MenuItem ? (
-      React.createElement(uiToReactComponent(notification.MenuItem))
+      <span key={notification.id}>
```

Comment:
I didn't try, but shouldn't you be able to pass `key={notification.id}` as part of `React.createElement` call?

Reply:
Good call, this worked perfectly!